specs/capri/FileShare_Design.md

```diff
@@ -0,0 +1,741 @@
+
+ “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+ “tenantId” : “string”
+
+ “userId” : “string“
+
+ “name” : “string”
+
+ “description” : “string”
+
+ “size” : “int64”
+
+ “availabilityZone”: “string”
+
+ “status”: “string”
+
+ “profileId”: “string”
+
+ “poolId”: “string”
+
+ “snapshotId”: “string”
+```
+##### 5. 	DELETE /v1beta/{tenantId}/file/shares/{shareId}
+* Deletes a share
+##### Request
+```json
+    “id” : “string”
+```
+##### 6.	GET /v1beta/{tenantId}/file/snapshots
+* Lists all share snapshots
+#### Request
+```json
+  “tenantId”: “string”
+```
+ ##### Response parameters
+```json
+ “id”: “string”
+
+ “protocols” : "[]string"
+
+ “createdAt” : “2019-03-20T12:49:00.497Z”
+
+ “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+ “name” : “string”
+
+ “description” : “string”
+ 
+ "shareId" : "string"
+
+ “shareSize” : ”int64”
+
+ “snapshotSize” : “int64”
+
+ “status”: “string”
+```
+##### 7.	GET /v1beta/{tenantId}/file/snapshots/{snapshotId}
+* Show details of snapshot
+##### Request
+```json
+  “id”: “string”
+```  
+##### Response parameters
+```json
+  “id”: “string”
+  
+  “shareId” : “string“
+
+  “protocols” : "[]string"
+
+  “createdAt” : “2019-03-20T12:49:00.497Z”
+
+  “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+  “name” : “string”
+
+  “description” : “string”
+
+  “shareSize” : “int64”
+
+  “snapshotSize” : “int64”
+
+  “status”: “string”
+```
+##### 8.   POST /v1beta/{tenantId}/file/snapshot
+* Creates a snapshot from a share.
+##### Request
+```json
+  “description” : “string”
+
+  “name” : “string”
+
+  “shareId” : “string“
+```
+  ##### Response parameters
+```json  
+  “id”: “string”
+
+  “shareId” : “string”
+
+  “protocols” : "[]string"
+
+  “createdAt” : “2019-03-20T12:49:00.497Z”
+
+  “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+  “name” : “string”
+
+  “description” : “string”
+
+  “shareSize” : “int64”
+
+  “snapshotSize” : “int64”
+
+  “status”: “string”
+```
+##### 9.	PUT /v1beta/{tenantId}/file/snapshots/{snapshotId}
+* Updates a share snapshot.
+##### Request
+```json
+  “description” : “string”
+
+  “name” : “string”
+```  
+##### Response parameters
+```json
+  “id”: “string”
+
+  “shareId” : “string”
+
+  “protocols” : "[]string"
+
+  “createdAt” : “2019-03-20T12:49:00.497Z”
+
+  “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+  “name” : “string”
+
+  “description” : “string”
+
+  “shareSize” : “int64”
+
+  “snapshotSize” : “int64”
+
+  “status”: “string”
+```
+##### 10.	DELETE /v1beta/{tenantId}/file/snapshots/{snapshotId}
+* Deletes a share
+##### Request
+```json
+  “id” : “sting”
+```
+##### 11.	POST /v1beta/{tenantId}/file/accesses/
+* Add access control for file share.
+##### Request
+```json
+    “sharetId” : “sting”
+
+    “type” : “string”    // user/ip
+
+    “accessCapability” : read/write/execute
+
+    “accessTo” : “user_list”
+```
+##### 12.	PUT /v1beta/{tenantId}/file/accesses/{accessId}
+* Add access control for file share.
+##### Request
+```json
+   “id” : “string”
+```
```

Comment:
There should also be an interface for querying Access，right?

Reply:
You are right. There will an interface for query the Access.